.travis.yml

```diff
@@ -1,33 +1,39 @@
 language: c
 compiler:
 - clang
-- gcc
 sudo: false
 cache: ccache
 
-script:
-- tests/scripts/recursion.pl library/*.c
-- tests/scripts/check-generated-files.sh
-- tests/scripts/check-doxy-blocks.pl
-- tests/scripts/check-names.sh
-- tests/scripts/check-files.py
-- tests/scripts/doxygen.sh
-- cmake -D CMAKE_BUILD_TYPE:String="Check" .
-- make
-- make test
-- programs/test/selftest
-- OSSL_NO_DTLS=1 tests/compat.sh
-- tests/ssl-opt.sh -e '\(DTLS\|SCSV\).*openssl'
-- tests/scripts/test-ref-configs.pl
-- tests/scripts/curves.pl
-- tests/scripts/key-exchanges.pl
+jobs:
+  include:
+    - name: sanity
+      script:
+        - tests/scripts/all.sh 'check_*'
+    - name: default configuration
+      script:
+        - cmake -D CMAKE_BUILD_TYPE:String="Check" .
+        - make
+        - make test
+        - programs/test/selftest
+        - OSSL_NO_DTLS=1 tests/compat.sh
+        - tests/ssl-opt.sh -e '\(DTLS\|SCSV\).*openssl'
```

Comment:
Assuming DTLS now works with the OpenSSL version on Travis, we could consider replacing this entire list with just `tests/scripts/all.sh test_default_cmake_gcc_asan`:
- easier maintainance, when things get added (such as `context-info.sh`) we get them here for free
- ASan
- (very marginal) one build with GCC in addition to the others with Clang.

Reply:
> when things get added (such as context-info.sh) we get them here for free
Another example: `tests/scripts/run_demos.py`